template/parse.go

```diff
@@ -242,6 +243,36 @@ func (r *rawTemplate) Template() (*Template, error) {
 		result.Provisioners = append(result.Provisioners, &p)
 	}
 
+	// Gather the error-cleanup-provisioner
+	if r.CleanupProvisioner != nil {
+		var p Provisioner
```

Comment:
I think the body of this `for` could be refactored into a `func` to decode provisioners. So that bugs come and go only once.